cardano-cli/src/Cardano/CLI/Run/Friendly.hs

```diff
@@ -11,156 +11,156 @@
-friendlyTxBodyBS era =
-  encodePretty (setConfCompare compare defConfig) . friendlyTxBody era
-
-friendlyTxBody :: CardanoEra era -> Api.TxBody era -> Value
-friendlyTxBody era txbody =
-  Object $
-    HashMap.fromList ["era" .= toJSON era]
-    <>
-    case txbody of
-      ByronTxBody body -> friendlyTxBodyByron body
-      ShelleyTxBody ShelleyBasedEraShelley body aux ->
-        addAuxData aux $ friendlyTxBodyShelley body
-      ShelleyTxBody ShelleyBasedEraAllegra body aux ->
-        addAuxData aux $ friendlyTxBodyAllegra body
-      ShelleyTxBody ShelleyBasedEraMary body aux ->
-        addAuxData aux $ friendlyTxBodyMary body
-
-addAuxData :: Show a => Maybe a -> Object -> Object
-addAuxData = HashMap.insert "auxiliary data" . maybe Null (toJSON . textShow)
-
-friendlyTxBodyByron :: Annotated Byron.Tx ByteString -> Object
-friendlyTxBodyByron = assertObject . toJSON
-
-friendlyTxBodyShelley
-  :: Shelley.TxBody (Ledger.ShelleyEra StandardCrypto) -> Object
-friendlyTxBodyShelley body =
-  HashMap.fromList
-    [ "inputs" .= Shelley._inputs body
-    , "outputs" .= fmap friendlyTxOutShelley (Shelley._outputs body)
-    , "certificates" .= fmap textShow (Shelley._certs body)
-    , "withdrawals" .= Shelley.unWdrl (Shelley._wdrls body)
-    , "fee" .= Shelley._txfee body
-    , "time to live" .= Shelley._ttl body
-    , "update" .= fmap textShow (Shelley._txUpdate body)
-    , "metadata hash" .= fmap textShow (Shelley._mdHash body)
+friendlyTxBodyBS
+  :: IsCardanoEra era => Api.TxBody era -> Either Prelude.String ByteString
+friendlyTxBodyBS =
+  fmap (encodePretty $ setConfCompare compare defConfig) . friendlyTxBody
+
+friendlyTxBody
+  :: forall era
+  .  IsCardanoEra era => Api.TxBody era -> Either Prelude.String Value
+friendlyTxBody txbody =
+  case getTransactionBodyContent txbody of
+    Right
+      TxBodyContent
+        { txIns
+        , txOuts
+        , txFee
+        , txValidityRange = txValidityRange@(_, upperBound)
+        , txMetadata
+        , txAuxScripts
+        , txWithdrawals
+        , txCertificates
+        , txUpdateProposal
+        , txMintValue
+        } ->
+      Right $
+      object
+        $   [ "era"     .= era
+            , "fee"     .= friendlyFee txFee
+            , "inputs"  .= txIns
+            , "outputs" .= map friendlyTxOut txOuts
+            ]
+        ++  [ "auxiliary scripts" .= friendlyAuxScripts txAuxScripts
+            | Just _ <- [auxScriptsSupportedInEra era]
+            ]
+        ++  [ "certificates" .= friendlyCertificates txCertificates
+            | Just _ <- [certificatesSupportedInEra era]
+            ]
+        ++  [ "metadata" .= friendlyMetadata txMetadata
+            | Just _ <- [txMetadataSupportedInEra era]
+            ]
+        ++  [ "mint" .= friendlyMintValue txMintValue
+            | Right _ <- [multiAssetSupportedInEra era]
+            ]
+        ++  [ "update proposal" .= friendlyUpdateProposal txUpdateProposal
```

Comment:
Here and in other places we are making backwards incompatible changes to the output structure. Are we OK with this because it is just a "friendly" output?

Reply:
Yes, this is just UI. API compatibility rules don't apply. I think, now the name is more correct.

Reply:
@cblp Please check with @disassembler and anyone else that may be using the pretty rendering to be sure.